keyleak/access_control.py

```
@@ -19,17 +19,38 @@
 
 
+def _guarded_requests_get(url: str, **kwargs: Any) -> Any:
+    """Default fetch: SSRF-guard the (crawl-derived, attacker-influenceable)
+    target host before issuing a real request. Raising a ``RequestException`` for
+    a blocked host makes the caller skip the URL via its existing handler."""
+    from .net_guard import url_block_reason
+
+    reason = url_block_reason(url)
+    if reason:
+        raise requests.RequestException(f"SSRF guard refused target: {reason}")
+    return requests.get(url, **kwargs)
```

Ruff (0.15.15)

[error] Probable use of `requests` call without timeout

(S113)